### PR TITLE
[ OC-1044 ]  Add export function

### PR DESCRIPTION
**Pull request by taoufikbermaki**
Please read through the checklist below to make sure that you haven't forgotten anything before submitting your PR.
Then, remove the text and replace it with the information and comments that should go into the release notes.

// tag::review_checklist[]
* The PR comment contains a brief explanation of the impact of these developments, and a pointer
to the relevant documentation if need be. Otherwise, it should why this development doesn't need
to be on the release notes.
* Check that Travis build is passing (for both branch and PR)
* Check that the base branch (i.e. the branch into which we want to merge changes) is correct: for feature branches
pull requests, this branch should be `develop`.
* Look through changed files to make sure everything is relevant to the PR (no mistakenly added changes,
no secret information, no malicious changes) and to see if you have remarks on the way things are implemented
* Check that the commit(s) message(s) is(are) relevant and follow conventions ([OC-XXX] ... )
* If there is more than one commit, is it meaningful or do we need to squash ?
* Meaningful and sufficient unit tests for the backend (80% coverage is the threshold)
* Meaningful unit tests for the frontend (Angular tests can be complex to implement, we should focus on testing complex
logic and not the framework itself)
* Check Sonar analysis on SonarCloud to see if Sonar reports any bugs or code smells on the new code
* API testing via Karate has been updated
* Cypress UI tests have been updated
* Documentation has been updated (especially if configuration is needed)
* Configuration examples have been updated (both for dev and docker)
* Build and run OpFab locally to see the new feature or bug fix at work. In the case of a new feature, it's also a way
of making sure that the configuration documentation is correct and easily understandable.
* Depending on the scope of the PR , build docker images and test in docker mode
* Check that the copyright header has been updated on the changed files if need be, and in the case of a first-time
    contributor, make sure they're added to the AUTHORS.txt file.
// end::review_checklist[]

### CODE REVIEW

File: ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts [hunk 1]
```diff
@@ -7,30 +7,89 @@
...
+        });
+
+        const filters = new Map<string, string[]>();
+        filters.set('size', [totalElements]);
+
+        this.cardService.fetchLoggingResults(filters).pipe(takeUntil(this.unsubscribe$))
+            .subscribe((page: Page<LineOfLoggingResult>) => {
+                const lines = page.content;
+
+                Array.prototype.forEach.call(lines, (line: LineOfLoggingResult) => {
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts:65]: `Array.prototype.forEach.call(lines, ... `

replace with : 
  `lines.forEach( (line: LineOfLoggingResult) => ...  `

File: ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts [hunk 2]
```diff
@@ -1,25 +1,80 @@
...
+    ngOnInit() {
+        this.initExportMonitoringData();
+
+    }
+
+    initExportMonitoringData(): void {
+
+        let time: string, businessPeriod: string, processName: any, title: any, summary: any, status: any;
+
+        Array.prototype.forEach.call(this.result, (line: LineOfMonitoringResult) => {
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts:41]: `Array.prototype.forEach.call(this.result, ...`

replace with :
`this.result.forEach( (line: LineOfMonitoringResult) => ...
`

**Comment by freddidierRTE** [comment]
- The data for the export are set at component startup but the export should reflect the result of the search . With this code , if we make a new search , the export is still done with the initial search. 
- For the monitoring screen , the data are update in realtime,  the export should take the last version.
- Correct sonar warning : https://sonarcloud.io/project/issues?id=org.lfenergy.operatorfabric%3Aoperatorfabric-core&pullRequest=492&resolved=false

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts (hunk 1 above)
Comment (freddidierRTE) [reply ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts:73]: See https://stackoverflow.com/questions/43743560/foreach-vs-array-prototype-foreach-call

**Comment by freddidierRTE** [comment]
Still one sonar issue : https://sonarcloud.io/project/issues?branch=OC-1044&id=org.lfenergy.operatorfabric%3Aoperatorfabric-core&resolved=false

File: ui/main/package.json [hunk 3]
```diff
@@ -58,9 +59,11 @@
     "ngrx-router": "^2.0.1",
     "ngx-countdown": "^10.0.0",
     "ngx-type-ahead": "^2.0.1",
-    "rxjs": "^6.6.3",
+    "rxjs": "^6.5.5",
```
[the remark this answers is not in the document]
Comment (taoufikbermaki) [reply ui/main/package.json]: ok

File: ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts (hunk 1 above)
Comment (taoufikbermaki) [reply ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts:73]: OK

File: ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts (hunk 2 above)
Comment (taoufikbermaki) [reply ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts:51]: ok

Comment (taoufikbermaki) [reply ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts:51]: ok

File: ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts (hunk 1 above)
Comment (taoufikbermaki) [reply ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts:73]: ok

File: ui/main/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (taoufikbermaki) [reply ui/main/package.json]: ok

File: ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts [hunk 4]
```diff
@@ -7,30 +8,99 @@
...
+    exportLoggingData: Array<any> ;
+    unsubscribe$: Subject<void> = new Subject<void>();
+
+
+    constructor(public timeService: TimeService, private cardService: CardService,
+        private store: Store<AppState>, private translate: TranslateService, private exportService: ExportService) {
+    }
 
-    constructor(public timeService: TimeService) {
+    ngOnChanges() {
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/logging/components/logging-table/logging-table.component.ts:44]: Do not call ExportMonitoringData it for each change and on it , call it just when user want to export (in the export method)

**Pull request merged**